q_agent_func_approx.py

```diff
@@ -30,7 +30,40 @@ def __init__(self, observation_space, action_space, **userconfig):
 
         # TODO define Q_opt as self.weights dotproduct phi(s,a) (similar to get_Qopt in blackjack 4a assignment)
         self.q = defaultdict(lambda: self.config["init_std"] * numpy.random.randn(self.action_n) + self.config["init_mean"])
+    def getQ(self, state, action):
+        score = 0
+        for f, v in self.feature_extractor(state, action):
+            score += self.weights[f] * v
+        return score
 
+
+
+    # Call this function to get the step size to update the weights.
+    def getStepSize(self):
```

Comment:
maybe we should try first with smaller step size and then see if a decreasing one would get better results